container/attribute.go

```diff
@@ -77,24 +77,25 @@ func (a Attributes) ToV2() []container.Attribute {
 
 // sets value of the attribute by key.
 func setAttribute(c *Container, key, value string) {
-	var a *Attribute
-
-	iterateAttributes(c, func(a_ *Attribute) bool {
-		if a_.Key() == key {
-			a = a_
+	attrs := c.Attributes()
+	found := false
+
+	for i := range attrs {
+		if attrs[i].Key() == key {
+			attrs[i].SetValue(value)
+			found = true
+			break
 		}
+	}
 
-		return a != nil
-	})
-
-	if a == nil {
-		a = NewAttribute()
+	if !found {
+		a := *NewAttribute()
 		a.SetKey(key)
-
-		c.SetAttributes(append(c.Attributes(), *a))
+		a.SetValue(value)
+		attrs = append(attrs, a)
```

Comment:
We use this pattern in api-go, should save us an allocation and attribute copying.
```
index := len(attrs)
attrs = append(attrs, Attribute{})
attrs[index].SetKey(key)
attrs[index].SetKey(value)
```